service/src/test/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumerTest.kt

```diff
@@ -423,7 +446,286 @@ class EventStreamConsumerTest {
+            Assertions.assertEquals(MarkerTransferRecord.find { MTT.txHash eq txHash }.count(), 1)
+        }
+    }
+
+    @Test
+    fun `recipient is not the member bank instance, does not persist, does not process`() {
+        val txHash = randomTxHash()
+        val transfer = getTransferEvent(txHash, "invalidrecipient", denom = serviceProperties.dccDenom)
+
+        whenever(pbcServiceMock.managerAddress).thenReturn(TEST_MEMBER_ADDRESS)
+
+        eventStreamConsumer.handleEvents(
+            blockHeight = 50,
+            transfers = listOf(transfer),
+            mints = listOf(),
+            burns = listOf(),
+            redemptions = listOf()
         )
+
+        verify(pbcServiceMock, never()).getTransaction(any())
+
+        transaction {
+            Assertions.assertEquals(TxStatusRecord.find { TST.txHash eq txHash }.count(), 0)
+            Assertions.assertEquals(MarkerTransferRecord.find { MTT.txHash eq txHash }.count(), 0)
+        }
+    }
+
+    @Test
+    fun `recipient is the member bank instance, denom is not valid, does not persist, does not process`() {
+        val txHash = randomTxHash()
+        val transfer = getTransferEvent(txHash, denom = "invaliddenom")
+
+        whenever(pbcServiceMock.managerAddress).thenReturn(TEST_MEMBER_ADDRESS)
+
+        eventStreamConsumer.handleEvents(
+            blockHeight = 50,
+            transfers = listOf(transfer),
+            mints = listOf(),
+            burns = listOf(),
+            redemptions = listOf()
+        )
+
+        verify(pbcServiceMock, never()).getTransaction(any())
+
+        transaction {
+            Assertions.assertEquals(TxStatusRecord.find { TST.txHash eq txHash }.count(), 0)
+            Assertions.assertEquals(MarkerTransferRecord.find { MTT.txHash eq txHash }.count(), 0)
+        }
+    }
+
+    @Test
+    fun `valid transfer persists, processes`() {
+        val txHash = randomTxHash()
+        val transfer = getTransferEvent(txHash, toAddress = TEST_MEMBER_ADDRESS, denom = serviceProperties.dccDenom)
+
+        whenever(pbcServiceMock.managerAddress).thenReturn(TEST_MEMBER_ADDRESS)
+        val txResponseSuccess = getDefaultTransactionResponse(txHash)
+        whenever(pbcServiceMock.getTransaction(any())).thenReturn(txResponseSuccess)
+
+        eventStreamConsumer.handleEvents(
+            blockHeight = 50,
+            transfers = listOf(transfer),
+            mints = listOf(),
+            burns = listOf(),
+            redemptions = listOf()
+        )
+
+        verify(pbcServiceMock).getTransaction(txHash)
+
+        transaction {
+            Assertions.assertEquals(TxStatusRecord.find { TST.txHash eq txHash }.count(), 0)
+            Assertions.assertEquals(
+                MarkerTransferRecord.find {
+                    (MTT.txHash eq txHash) and (MTT.status eq MarkerTransferStatus.INSERTED)
+                }.count(),
+                1
+            )
+        }
+    }
+
+    @Test
```

Comment:
would some of these tests benefit from inner classes to scope them relative to a specific type of event or purpose?